packages/web/lib/hooks/players.ts

```diff
@@ -120,7 +139,74 @@ export const useFiltersUsed = (
       timezonesFilterUsed,
     ],
   );
+
+  return filtersUsed;
+};
+
+const usePaginatedPlayers = (
+  queryVariables: GetPlayersQueryVariables,
+  setQueryVariable: QueryVariableSetter,
+) => {
+  const {
+    fetching,
+    players: fetchedPlayers,
+    totalCount,
+    error,
+  } = useFilteredPlayers(queryVariables);
+
+  const itemsPerPage = PLAYER_LIMIT;
+  const maxPage = Math.ceil(totalCount / itemsPerPage);
+  const currentOffset = useMemo(() => (queryVariables.offset as number) || 0, [
+    queryVariables.offset,
+  ]);
+  const currentPage = useMemo(
+    () => Math.ceil(currentOffset / itemsPerPage) + 1,
+    [currentOffset, itemsPerPage],
+  );
+
+  const shouldAppend = useRef(false);
+
+  const nextPage = useCallback(() => {
+    if (currentPage < maxPage && !fetching && !shouldAppend.current) {
+      shouldAppend.current = true;
+      setQueryVariable('offset', currentOffset + itemsPerPage);
+    }
+  }, [
+    shouldAppend,
+    fetching,
+    setQueryVariable,
+    currentOffset,
+    itemsPerPage,
+    currentPage,
+    maxPage,
+  ]);
+
+  const [players, setPlayers] = useState<PlayerFragmentFragment[]>(
+    fetchedPlayers,
+  );
+
+  useEffect(() => {
+    if (error || fetching) {
+      if (error || !shouldAppend.current) {
+        setPlayers([]);
+      }
+      return;
+    }
+    if (shouldAppend.current) {
+      setPlayers((_players) => [..._players, ...fetchedPlayers]);
```

Comment:
```suggestion
      setPlayers((players) => [...players, ...fetchedPlayers]);
```

Reply:
Generally an underscore in front of a variable means it isn't referenced in the body of the function.

Reply:
I can't set it as `players` since that variable is used outside. But maybe some other variable name like `oldPlayers`